src/betterproto/__init__.py

```diff
@@ -455,20 +455,19 @@ def _dump_float(value: float) -> Union[float, str]:
 
     Parameters
     ----------
-    value : float
+    value: float
         Value to dump
 
     Returns
     -------
     Union[float, str]
-        Dumped valid, either a float or the strings
-        "Infinity" or "-Infinity"
+        Dumped value, either a float or the strings
     """
     if value == float("inf"):
         return INFINITY
     if value == -float("inf"):
         return NEG_INFINITY
-    if value == float("nan"):
+    if math.isnan(value):
```

Comment:
```suggestion
    if isinstance(value, float) and math.isnan(value):
```

I am not sure why the test case ended up passing a `None` value in the first place when this method does not expect it.